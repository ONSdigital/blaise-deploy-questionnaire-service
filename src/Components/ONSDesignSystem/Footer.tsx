import React, {ReactElement} from "react";

function Footer(): ReactElement {
    return (
        <>
            <footer className="footer">
                <div className="footer__body" data-analytics="footer">
                    <div className="container ">
                        <div className="grid grid--flex grid--between">
                            <div className="grid__col">
                                <div className="u-mt-m u-mb-s">
                                    <img src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                                         alt="Office for National Statistics"/>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </footer>
        </>
    );
}

export default Footer;
