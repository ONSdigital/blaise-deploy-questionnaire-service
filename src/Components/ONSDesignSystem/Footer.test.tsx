import React from "react";
import Enzyme, {render, shallow} from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import Footer from "./Footer";

describe("Footer Test", () => {
    Enzyme.configure({adapter: new Adapter()});

    it("matches Snapshot", () => {
        const wrapper = render(<Footer/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const wrapper = shallow(<Footer/>);
        expect(wrapper.exists()).toEqual(true);
    });
});
