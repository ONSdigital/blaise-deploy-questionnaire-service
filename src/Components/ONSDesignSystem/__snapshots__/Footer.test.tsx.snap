// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Test matches Snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "footer",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "footer__body",
          "data-analytics": "footer",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "container ",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "grid grid--flex grid--between",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "grid__col",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "u-mt-m u-mb-s",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "alt": "Office for National Statistics",
                              "src": "https://ons-design-system.netlify.app/img/ons-logo-black-en.svg",
                            },
                            "children": Array [],
                            "name": "img",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "alt": undefined,
                              "src": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "alt": undefined,
                              "src": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-analytics": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-analytics": undefined,
        },
      },
    ],
    "name": "footer",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
